src/message/messagesActions.js

```diff
@@ -104,16 +111,21 @@ export const fetchNewer = () => (dispatch: Dispatch, getState: GetState): Action
   }
 };
 
-export const doNarrow = (newNarrow: Narrow, anchor: number = Number.MAX_SAFE_INTEGER): Action => (
+export const doNarrow = (newNarrow: Narrow, anchorId?: number): Action => (
   dispatch: Dispatch,
   getState: GetState,
 ) => {
   const auth = getAuth(getState());
   const anyMessagesInNewNarrow = JSON.stringify(newNarrow) in getAllMessages(getState());
-
-  if (!anyMessagesInNewNarrow) {
+  if (!anyMessagesInNewNarrow && !anchorId) {
     dispatch(fetchMessagesAtFirstUnread(newNarrow));
   }
   registerAppActivity(auth);
-  dispatch(switchNarrow(newNarrow));
+
+  if (anchorId) {
+    const messages = getMessagesInActiveNarrow(getState());
+    dispatch(switchNarrow(newNarrow, messages && messages.find(msg => msg.id === anchorId)));
```

Comment:
i think it isn't possible, I added this check for local testing, forgot to remove it